main.py

```diff
@@ -78,124 +38,34 @@ def process_keyboard_events(actions_while_key_pressed, actions_on_key_down):
-        colors_dict["blue"]
-    )
-
-    toggle_overlay_visibility_on_down = lambda: overlays[0].toggle_visibility()
-
-    activate_ego_upwards_boost = lambda: ego.kinematic.external_forces.append(
-        CommonConstants.ROCKET_UPWARD_BOOST_FORCE_SCALAR * Vec2d(0, -1)
-    )
-    activate_left_engine_boost = lambda: ego.kinematic.external_forces.append(
-        CommonConstants.ROCKET_SIDEWAYS_BOOST_FORCE_SCALAR * Vec2d(-1, 0)
-    )
-    activate_right_engine_boost = lambda: ego.kinematic.external_forces.append(
-        CommonConstants.ROCKET_SIDEWAYS_BOOST_FORCE_SCALAR * Vec2d(1, 0)
-    )
-
-    act_change_box_color_while_spacebar_pressed = LandingGameActionOnKey(
-        PygameKeyState(pygame.K_SPACE, True), color_key_indicator_blue_while_pressed
-    )
-    act_change_box_color_while_spacebar_not_pressed = LandingGameActionOnKey(
-        PygameKeyState(pygame.K_SPACE, False), color_key_indicator_cyan_while_pressed
-    )
-
-    act_change_box_color_on_spacebar_down = LandingGameActionOnKey(
-        PygameKeyState(pygame.K_SPACE, True), color_key_indicator_blue_on_down
-    )
-
-    act_toggle_overlay_visibility_on_v_down = LandingGameActionOnKey(
-        PygameKeyState(pygame.K_v, True), toggle_overlay_visibility_on_down
-    )
-
-    act_boost_ego_up_on_upwards_key = LandingGameActionOnKey(
-        PygameKeyState(pygame.K_UP, True), activate_ego_upwards_boost
-    )
-    act_boost_ego_left_on_left_key = LandingGameActionOnKey(
-        PygameKeyState(pygame.K_LEFT, True), activate_left_engine_boost
-    )
-    act_boost_ego_up_on_rigth_key = LandingGameActionOnKey(
-        PygameKeyState(pygame.K_RIGHT, True), activate_right_engine_boost
-    )
-
-    obj_list = LandingGameGroup()
-    obj_list.add(key_indicator_while_pressed)
-    obj_list.add(key_indicator_on_down)
-    obj_list.add(ego)
-    obj_list.add(ground)
-    for overlay in overlays:
-        obj_list.add(overlay)
-
-    # bundle all keystate -> action correlations into one list
-    actions_while_key_pressed = [
-        act_change_box_color_while_spacebar_pressed,
-        act_change_box_color_while_spacebar_not_pressed,
-        act_boost_ego_up_on_upwards_key,
-        act_boost_ego_left_on_left_key,
-        act_boost_ego_up_on_rigth_key,
-    ]
-    actions_on_key_down = [
-        act_change_box_color_on_spacebar_down,
-        act_toggle_overlay_visibility_on_v_down,
-    ]
+    scenario = Scenario()
+    overlays = ScenarioOverlays(scenario, game_timing)
 
     while True:
-        process_keyboard_events(actions_while_key_pressed, actions_on_key_down)
+        process_each_frame_events(scenario.actions_each_frame)
+        process_keyboard_events(
+            scenario.actions_while_key_pressed, scenario.actions_on_key_down
+        )
         game_window.erase_screen()
-        ego.kinematic.external_forces.append(
+        scenario.object_list.get_object_by_name("ego").kinematic.external_forces.append(
```

Comment:
option to increase readability:
split this in two lines. first `ego_object = scenraio.object_list.get_object_by_name("ego")`

and second is appending the forces to `ego_obejct`